root/account/applications/Index.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {compare, l} from '../../static/scripts/common/i18n';
+import Layout from '../../layout';
+import PaginatedResults from '../../components/PaginatedResults';
+import commaOnlyList from '../../static/scripts/common/i18n/commaOnlyList';
+import loopParity from '../../utility/loopParity';
+
+type Props = {|
+  +applications: $ReadOnlyArray<ApplicationT>,
+  +appsPager: PagerT,
+  +tokens: $ReadOnlyArray<EditorOAuthTokenT>,
+  +tokensPager: PagerT,
+|};
+
+const buildApplicationRow = (application: ApplicationT, index: number) => (
+  <tr className={loopParity(index)} key={application.id}>
+    <td>{application.name}</td>
+    <td>
+      {application.oauth_redirect_uri ? l('Web Application') : l('Installed Application')}
+    </td>
+    <td><code>{application.oauth_id}</code></td>
+    <td><code>{application.oauth_secret}</code></td>
+    <td>
+      <a href={'/account/applications/edit/' + application.id}>{l('Edit')}</a>
+      {' | '}
+      <a href={'/account/applications/remove/' + application.id}>{l('Remove')}</a>
+    </td>
+  </tr>
+);
+
+const buildTokenRow = (token: EditorOAuthTokenT, index: number) => (
+  <tr className={loopParity(index)} key={token.id}>
+    <td>{token.application.name}</td>
+    <td>{formatScopes(token)}</td>
+    <td>
+      <a href={'/account/applications/revoke-access/' + token.application.id + '/' + token.scope}>
+        {l('Revoke Access')}
+      </a>
+    </td>
+  </tr>
+);
+
+function formatScopes(token: EditorOAuthTokenT) {
+  const lScopes = [];
+
+  token.permissions.forEach((perm) => {
```

Comment:
You can initialize the list with a map like `const lScopes = token.permissions.map(perm => l(perm));` to be slightly more expressive

Reply:
Done at https://github.com/metabrainz/musicbrainz-server/pull/699/commits/79ae598a8cec954a67da07209377d94c74ae508a#diff-4c3d028cef4917d20675baed1fb0ce28R54